scipy/signal/fir_filter_design.py

```diff
@@ -30,6 +30,24 @@ def _get_fs(fs, nyq):
     return fs
 
 
+def _validate_fs(fs, allow_none=True):
+    """
+    Check if the given sampling frequency is a scalar and raises an exception
+    otherwise. If allow_none is False, also raises an exception for none
+    sampling rates. Returns the sampling frequency as float or none if the
+    input is none.
+    """
+    if not allow_none and fs is None:
+        raise ValueError("Sampling frequency can not be none.")
+    if fs is not None and not np.isscalar(fs):
+        raise ValueError("Sampling frequency fs must be a single scalar.")
+
+    if fs is None:
+        return None
+    else:
+        return float(fs)
```

Comment:
Having two copies of this violates DRY, I would put one copy in one place that both files can import from. `_arraytools` would be one option if you can't find a better one

Reply:
Thanks for pointing this out. I moved the function to _arraytools.